lib/game.rb

```diff
@@ -1,56 +1,59 @@
 require_relative 'board'
-require_relative 'console'
 require_relative 'rules'
+require_relative 'human_player'
+require_relative 'dumb_player'
 
 class Game
-  attr_accessor :board, :console, :rules
+  attr_accessor :board, :rules, :player_one, :player_two
 
-  def initialize(board, console, rules)
+  def initialize(board, rules, display, player_creator)
     @board = board
-    @console = console
-    @current_player = 'X'
     @rules = rules
+    @display = display
+    @player_creator = player_creator
+    @current_player = @player_one
   end
 
-  def welcome
-    console.print_message('Welcome to Tic Tac Toe')
+  def start
+    @display.welcome
+    player_setup
+    play
   end
 
   def turn
-    alert_current_player
-    retrieve_user_move
-    console.print_message(@board.display)
+    @display.alert_current_player(@current_player)
+    @board.mark(@current_player.move, @current_player.mark)
+    @display.board
     switch_players
   end
 
-  def retrieve_user_move
-    console.print_message("Player #{@current_player}, please enter a position 1-9 that is not already marked")
-    index = @console.retrieve_user_input.to_i - 1
-    board.valid_move?(index) ? @board.mark(index, @current_player) : retrieve_user_move
-  end
-
-  def switch_players
-    if !rules.game_over?(board)
-      @current_player == 'X' ? @current_player = 'O' : @current_player = 'X'
-    end
+  def player_setup
+    @player_creator.create_players(@board, @display)
+    @player_one = @player_creator.player_one
+    @player_two = @player_creator.player_two
   end
 
   def play
+    @current_player = @player_one
```

Comment:
Why do you need this line? Isn't `@current_player` specified in the initializer?

Reply:
When the game is initialized the one in the initialize method is = nil. Because Ruby passes by value and not by reference, then it needs to be set again. I have deleted it from the initialize method now. Any reason this isnt a good idea?